test/py/CMakeLists.txt

```diff
@@ -23,37 +23,62 @@
         set (ENV_COMMAND ${CMAKE_COMMAND} -E env 
             "PYTHONPATH=$<TARGET_FILE_DIR:migraphx_pybind_${PYTHON_VERSION}>"
             "PYTHONMALLOC=debug"
             "MALLOC_CHECK_=3"
         )
         set(PYTHON_EXECUTABLE ${PYTHON_${PYTHON_VERSION}_EXECUTABLE})
+        if(${PYTHON_VERSION} STREQUAL "3.6")
+            add_py_venv_fixture(${PYTHON_VERSION} ${PYTHON_EXECUTABLE} ${VENV_DIR}/${PYTHON_VERSION} ${REQUIREMENTS_PYTHON_36})
+        else()
+            add_py_venv_fixture(${PYTHON_VERSION} ${PYTHON_EXECUTABLE} ${VENV_DIR}/${PYTHON_VERSION} ${REQUIREMENTS_PYTHON_ALL})
+        endif()
+        unset(PYTHON_EXECUTABLE)
+        set(PYTHON_EXECUTABLE ${VENV_DIR}/${PYTHON_VERSION}/bin/python)
         add_test(
             NAME test_py_${PYTHON_VERSION}_${NAME} 
             COMMAND ${ENV_COMMAND} ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${SCRIPT} ${ARGN})
+        set_tests_properties(test_py_${PYTHON_VERSION}_${NAME} PROPERTIES FIXTURES_REQUIRED ${PYTHON_VERSION}_VENV)
         add_custom_target(test_py_${PYTHON_VERSION}_${NAME} 
             COMMAND ${ENV_COMMAND} ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${SCRIPT} ${ARGN} 
             COMMENT "${PYTHON_EXECUTABLE} ${SCRIPT}")
-
     endforeach()
 endfunction()
 add_dependencies(tests migraphx_py)
 add_dependencies(check migraphx_py)
 
-add_py_test(ref test_cpu.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(save_load test_save_load.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(op test_op.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(shape test_shape.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(module_construct test_module_construct.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(literal test_literal.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(ref test_cpu.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(save_load test_save_load.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(op test_op.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(shape test_shape.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(module_construct test_module_construct.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(literal test_literal.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
 if(MIGRAPHX_ENABLE_GPU)
-add_py_test(gpu_offload test_gpu_offload.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(gpu test_gpu.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(array test_array.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(backend onnx_backend_test.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
-add_py_test(gpu_async test_gpu_async.py WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(gpu_offload test_gpu_offload.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(gpu test_gpu.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(array test_array.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
+add_py_test(backend onnx_backend_test.py ${VENV} WORKING_DIRECTORY ${TEST_ONNX_DIR})
```

Comment:
Onnx backend should be using a different virtual environment, and this should be disabled on python 3.6.

Reply:
I have disabled the onnx backend tests on python 3.6. I am open to a more elegant solution if you think we can have an alternate way to disable this.